src/Raven.Client/Documents/Session/AsyncSessionDocumentTimeSeries.cs

```diff
@@ -115,6 +122,33 @@ await ServeFromCacheOrGetMissingPartsFromServerAndMerge(from ?? DateTime.MinValu
             return rangeResult.Entries;
         }
 
+        private async Task FixRange<TTValues>(DateTime? from,
+            DateTime? to,
+            int start,
+            int pageSize,
+            TimeSeriesRangeResult<TTValues> rangeResult,
+            CancellationToken token = default)
+            where TTValues : TimeSeriesEntry
+        {
```

Comment:
Would it make sense to this on the server-side? and spare the extra call?

Reply:
I think yes. In that case I can fix the range for start also